Add Unified Runtime v0.11.8 source package

frenchwr opened the pull request:
This adds Unified Runtime to our PPA, using the current version we use to build the DPC++ compiler (v6.1.0) in the PPA. Note we will NOT use this version for building the current version of the compiler as it cannot properly autodetect build dependencies like this one; however, the upcoming DPC++ release is expected to resolve this autodetection issue, and we want all the build dependencies (like Unified Runtime) ready to go (hopefully with only minor version bump updates in the PPA).

frenchwr committed:
Remove unpublished version of unified runtime

frenchwr committed:
Add unified-runtime source package

hector-cao commented:
Great !

I like the packaging quality,

One observation I have:
 
1) AFAIU, the UR delivers the adapters for different backends (levelzero, opencl, cuda, AMD HIP),

<img width="741" height="419" alt="image" src="https://github.com/user-attachments/assets/a5aa1fb1-5582-4c6a-afd7-62d3b6511b6e" />

  We will need to enable some more configure switches to enable those libraries:
  
  UR_BUILD_ADAPTER_L0	Build the Level-Zero adapter	ON/OFF	OFF
  UR_BUILD_ADAPTER_OPENCL	Build the OpenCL adapter	ON/OFF	OFF

hector-cao reviewed (dismissed):
LGTM :rocket:

frenchwr commented:
@hector-cao Ah you're right on the adapters! Let me include those.

frenchwr committed:
WIP: build level-zero and opencl adapters.

This build works locally, but it fails on Launchpad because
it requires a git clone of the level-zero experimental headers
into the unified runtime source tree.

frenchwr committed:
Vendor experimental level-zero headers for LP build

frenchwr commented:
@hector-cao It took a lot of fighting with CMake and Launchpad, but the level-zero and opencl adapters are now included. I took a similar approach as you for the compiler and included a `orig-vendor` tarball with the experimental level-zero headers.

frenchwr committed:
Remove spurious patch files

hector-cao reviewed (approved):
TBH, I did not do a thougtful review for this PR but in broad outline, it looks amazing !

Things can be refined in the upcoming 6.2 release.
